### PR TITLE
use regular client instead of kubectl in scheduler predicate tests when checking/setting/cleanning taints/labels

### PR DESCRIPTION
**Pull request by kevin-wangzefeng**
The existing implementation in scheduler predicate tests uses kubectl to check/set/clean taints/labels on node, which makes the test very related to kubectl.

 This PR is to use regular client instead.

### CODE REVIEW

**Comment by kevin-wangzefeng** [comment]
@gmarek @davidopp PTAL

**Comment by kevin-wangzefeng** [comment]
For the change on taint-toleration test in #29003, I will send another PR to fix, as well as address https://github.com/kubernetes/kubernetes/pull/24134#discussion_r63794924 .

File: test/e2e/scheduler_predicates.go [hunk 1]
```diff
@@ -100,6 +100,21 @@ func cleanupPods(c *client.Client, ns string) {
 	}
 }
 
+func addOrUpdateLabelOnNode(c *client.Client, nodeName string, labelKey string, labelValue string) {
```
Comment (gmarek) [inline test/e2e/scheduler_predicates.go]: We can probably move it to framework/util

File: test/e2e/scheduler_predicates.go [hunk 2]
```diff
@@ -100,6 +100,21 @@ func cleanupPods(c *client.Client, ns string) {
...
+	patch := fmt.Sprintf(`{"metadata":{"labels":{"%s":"%s"}}}`, labelKey, labelValue)
+	err := c.Patch(api.MergePatchType).Resource("nodes").Name(nodeName).Body([]byte(patch)).Do().Error()
+	framework.ExpectNoError(err)
+}
+
+func expectNodeHasLabel(c *client.Client, nodeName string, labelKey string, labelValue string) {
```
Comment (gmarek) [inline test/e2e/scheduler_predicates.go]: also framework/util

**Comment by gmarek** [comment]
I did a quick scan and generally LGTM. I'll leave the proper review for @davidopp or @wojtek-t

**Comment by kevin-wangzefeng** [comment]
@gmarek review comments addressed, PTAL

**Comment by davidopp** [comment]
LGTM

Thanks!

**Comment by kevin-wangzefeng** [comment]
@davidopp squashed, PTAL

**Comment by gmarek** [comment]
Thanks! Our tests are way more stable now.

**Pull request merged**